sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,116 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
+        _value_check(mean > 0, "Mean must be positive")
+
+    def pdf(self, x):
+        return exp(-self.shape*(x - self.mean)**2 / (2*x*self.mean**2)) * sqrt(self.shape/((2*pi*x**3)))
+
+    def sample(self):
+        scipy = import_module('scipy')
+        if scipy:
+            from scipy.stats import invgauss
+            return invgauss.rvs(float(self.mean/self.shape), 0, float(self.shape))
+        else:
+            raise NotImplementedError('Sampling the Inverse Normal Distribution requires Scipy.')
+
+    def _standardNormal_cdf(self, x):
+        return erf(sqrt(2)*(x - S.One)/2)/2 + S.Half
+
+    def _cdf(self, x):
+        mean, shape = self.mean, self.shape
+
+        first_term = self._standardNormal_cdf(sqrt(shape/x) * ((x/mean - S.One)))
+        second_term = exp(2*shape/mean) * self._standardNormal_cdf(-sqrt(shape/x)*(x/mean + S.One))
+
+        return  first_term + second_term
+
+    def _characteristic_function(self, t):
+        mean, shape = self.mean, self.shape
+        return exp((shape/mean)*(1 - sqrt(1 - (2*mean**2*I*t)/shape)))
+
+    def _moment_generating_function(self, t):
```

Comment:
Can you add tests for this and `_characteristic_function` methods?

Reply:
I have added test for `_characteristic_function`. Will add another PR, in which I will add tests for all `_moment_generating_function`s